### ISSUE TITLE
Timeout for File/Folder checks

### ISSUE THREAD

**Issue by nicholasamorim** [aelsabbahy/goss#444]
Is is possible to add timeout for file/folder checks? 

The use case being when these are mounted on a NFS and the storage behind that might be broken for whatever reason.  In certain cases you can `ls` a folder and that would hang indefinitely in the foreground.

**Comment by aelsabbahy** [comment]
This seems like a good feature. Do you know of a good way to reproduce this in order to make sure the enhancement satisfies the requirement?

**Comment by buckeye43210** [comment]
Yes, I can test this enhancement for you (if still needed). I have a use case where I check for a string in a log file to determine if my applications has registered with a cloud host. The registration takes several seconds -- if I run "molecule verify" too soon, it fails. It usually passes on the second run.
